### PR TITLE
[rush] Clean up the usage of custom typings in rush-lib.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: apps/rush-lib/src/logic/npm/NpmPackage.ts [hunk 1]
```diff
@@ -89,13 +89,13 @@ export class NpmPackage extends BasePackage {
...
-    const npmPackage: readPackageTree.PackageNode = {
+    const npmPackage: readPackageTree.Node = {
       children: [],
-      error: undefined,
+      error: null, // eslint-disable-line @rushstack/no-null
       id: 0,
       isLink: false,
       package: packageJson,
-      parent: undefined,
+      parent: null, // eslint-disable-line @rushstack/no-null
```
Comment (octogonz) [inline apps/rush-lib/src/logic/npm/NpmPackage.ts]: Why is this null?

File: apps/rush-lib/src/utilities/Utilities.ts [hunk 2]
```diff
@@ -312,7 +312,7 @@ export class Utilities {
       maxLineLength = Utilities.getConsoleWidth();
     }
 
-    const wrap: (textToWrap: string) => string = wordwrap.soft(indent, maxLineLength);
+    const wrap: (textToWrap: string) => string = wordwrap(indent, maxLineLength, { mode: 'soft' });
```
Comment (octogonz) [inline apps/rush-lib/src/utilities/Utilities.ts]: Please test this change

File: apps/rush-lib/src/logic/npm/NpmPackage.ts (hunk 1 above)
Comment (iclanton) [reply apps/rush-lib/src/logic/npm/NpmPackage.ts]: The `readPackageTree.Node.parent` and `readPackageTree.Node.error` types are `readPackageTree.Node | null` and `Error | null` respectively.

File: apps/rush-lib/src/utilities/Utilities.ts (hunk 2 above)
Comment (iclanton) [reply apps/rush-lib/src/utilities/Utilities.ts]: The behavior is the same.

**Pull request merged**